src/hash_table.c

```diff
@@ -622,23 +492,11 @@ lyht_remove(struct ly_ht *ht, void *val_p, uint32_t hash)
 LIBYANG_API_DEF uint32_t
 lyht_get_fixed_size(uint32_t item_count)
 {
-    uint32_t i, size = 0;
-
-    /* detect number of upper zero bits in the items' counter value ... */
-    for (i = (sizeof item_count * CHAR_BIT) - 1; i > 0; i--) {
-        size = item_count << i;
-        size = size >> i;
-        if (size == item_count) {
-            break;
-        }
-    }
-    assert(i);
-
-    /* ... and then we convert it to the position of the highest non-zero bit ... */
-    i = (sizeof item_count * CHAR_BIT) - i;
+    unsigned int bits = sizeof(uint32_t) * 8;
 
-    /* ... and by using it to shift 1 to the left we get the closest sufficient hash table size */
-    size = 1 << i;
+    if (item_count == 1)
+        return 1;
 
-    return size;
+    /* return next power of 2 (greater or equal) */
+    return 1 << (bits - __builtin_clz(item_count - 1));
```

Comment:
You can optimize this, I have never attempted to, but using compiler built-ins is just asking for compatibility problems so please avoid it.

Reply:
I didn't think that it was not available under Windows. I'll fallback to something more standard.